pkg/server/bootstrap_server.go

```diff
@@ -102,6 +102,44 @@ func (bsc *bootstrapServer) GetConfig(cr poolRequest) (*runtime.RawExtension, er
 		return nil, fmt.Errorf("parsing Ignition config failed with error: %w", err)
 	}
 
+	fileName = path.Join(bsc.serverBaseDir, "controller-config", "machine-config-controller.yaml")
+	klog.Infof("reading file %q", fileName)
+	data, err = os.ReadFile(fileName)
+	if os.IsNotExist(err) {
+		klog.Errorf("could not find file: %s", fileName)
+		return nil, nil
+	}
+	if err != nil {
+		return nil, fmt.Errorf("server: could not read file %s, err: %w", fileName, err)
+	}
+
+	cc := new(mcfgv1.ControllerConfig)
+	err = yaml.Unmarshal(data, cc)
+	if err != nil {
+		return nil, fmt.Errorf("server: could not unmarshal file %s, err: %w", fileName, err)
+	}
+
+	existsKube := false
+	existsCloud := false
+	for idx, file := range ignConf.Storage.Files {
+		switch file.Path {
+		case caBundleFilePath:
+			existsKube = true
+			data := string(cc.Spec.KubeAPIServerServingCAData)
+			ignConf.Storage.Files[idx].Contents.Source = &data
+		case cloudProviderCAPath:
+			existsCloud = true
+			data := string(cc.Spec.CloudProviderCAData)
+			ignConf.Storage.Files[idx].Contents.Source = &data
+		}
+	}
+	if !existsKube {
+		appendFileToIgnition(&ignConf, caBundleFilePath, string(cc.Spec.KubeAPIServerServingCAData))
+	}
+	if !existsCloud {
+		appendFileToIgnition(&ignConf, cloudProviderCAPath, string(cc.Spec.CloudProviderCAData))
+	}
+
```

Comment:
i see some similar code for handling KubeAPIServerServingCAData and CloudProviderCAData in both bootstrap_server.go and cluster_server.go. maybe it can be a shared function or method? just a thought? a little nitty and not a requirement at all!